crates/core/src/operations/optimize.rs

```diff
@@ -624,18 +638,42 @@ impl MergePlan {
                     for file in files.iter() {
                         debug!("  file {}", file.path);
                     }
+
                     let object_store_ref = log_store.object_store(Some(operation_id));
+                    let file_format_options = file_format_options.clone();
                     let batch_stream = futures::stream::iter(files.clone())
                         .then(move |file| {
                             let object_store_ref = object_store_ref.clone();
                             let meta = ObjectMeta::try_from(file).unwrap();
+                            let file_format_options = file_format_options.clone();
                             async move {
+                                let decrypt: Option<FileDecryptionProperties> =
```

Comment:
We might be able to tidy this up by adding a method to get `ArrowReaderOptions`, or alternatively changing this to use DataFusion for reading like for Z-Order compaction.